### PR TITLE
feat: add immutable name snakecase rule

### PR DESCRIPTION
**Pull request by 0xRaccoon**
Signed-off-by: 0xRaccoon <raccoon@defi.sucks>

### CODE REVIEW

File: rules/immutable-name-snakecase.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    description: 'Immutable name must be in capitalized SNAKE_CASE.',
+    category: 'Style Guide Rules',
+  },
+  isDefault: true,
+  recommended: true,
+  defaultSetup: 'warn',
+  schema: null,
+}
+
+class ImmutableNameSnakecaseChecker extends BaseChecker {
```
Comment (gotzenx) [inline rules/immutable-name-snakecase.js]:
```suggestion
class ImmutableNameSnakeCaseChecker extends BaseChecker {
```

File: test/rules/immutable-name-snakecase.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import linter from '../utils/parsers';
+const contractWith = require('solhint/test/common/contract-builder').contractWith;
+
+const config = {
+    rules: { 'immutable-name-snakecase': 'error' },
+};
+
+describe('Linter - immutable-name-snakecase', () => {
+    it('should raise error for immutable name not in snakecase', () => {
+        const code = contractWith('uint256 immutable testVar;');
```
Comment (gotzenx) [inline test/rules/immutable-name-snakecase.ts]: use variable for name like in the other tests

File: test/rules/immutable-name-snakecase.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import assert from 'assert';
+import linter from '../utils/parsers';
+const contractWith = require('solhint/test/common/contract-builder').contractWith;
+
+const config = {
+    rules: { 'immutable-name-snakecase': 'error' },
+};
+
+describe('Linter - immutable-name-snakecase', () => {
+    it('should raise error for immutable name not in snakecase', () => {
```
Comment (0xGorilla) [inline test/rules/immutable-name-snakecase.ts:10]: what about immutable private, will `_TEST_VAR` throw an error or not?

Comment (0xRaccoon) [reply test/rules/immutable-name-snakecase.ts:10]: it shouldn't; good idea. I've just added the test for that scenario

**Pull request merged**